### PR TITLE
tests: Refactor vklayertests_best_practices

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Move ARM-specific tests into their own file.

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
This change simply takes all tests beginning with `VkArm` and moves them to their own file.

File: tests/vklayertests_arm_best_practices.cpp [hunk 1]
```diff
@@ -0,0 +1,945 @@
+/*
+ * Copyright (c) 2015-2020 The Khronos Group Inc.
+ * Copyright (c) 2015-2020 Valve Corporation
+ * Copyright (c) 2015-2020 LunarG, Inc.
+ * Copyright (c) 2015-2020 Google, Inc.
```
Comment (mark-lunarg) [inline tests/vklayertests_arm_best_practices.cpp:5]: You can safely remove this line.

File: tests/vklayertests_arm_best_practices.cpp [hunk 2]
```diff
@@ -0,0 +1,945 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Author: Camden Stocker <camden@lunarg.com>
```
Comment (mark-lunarg) [inline tests/vklayertests_arm_best_practices.cpp:13]: You can replace this with your name, por favor.

**Review by mark-lunarg — CHANGES_REQUESTED** [review]
This is awesome!  I'm guessing Android and the GN build worked because they're simply not building the new file -- search for `vklayertests_best_practices.cpp` in `BUILD.gn` and in the `build_android ` sub-tree to find out the other spots you need to modify, and then you can verify the ARM tests ran in the CI build artifacts.

**Review by mark-lunarg — COMMENTED** [review]
Sorry, one more thing -- please update the `BUILD.gn` file -- it's used for the Google Ninja builds I believe.

**Review by mark-lunarg — APPROVED** [review]

**Pull request merged**